lib/elixir_analyzer/exercise_test/common_checks/function_capture.ex

```diff
@@ -60,7 +60,7 @@ defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.FunctionCapture do
         functions
       end
 
-    {node, %{capture_depth: depth - 1, functions: functions}}
+    {node, %{capture_depth: depth, functions: functions}}
```

Comment:
How did you discover this bug? When I bring it back, no tests fail, so it wasn't by writing tests 🤔

Reply:
Sorry about that! I took me a while to remember.
I think it was from this test:

https://github.com/exercism/elixir-analyzer/blob/926d69d60f13ccee353520a09e9c882c53b0e4d2/test/elixir_analyzer/exercise_test/common_checks/function_capture_test.exs#L53-L58

At some point I was checking if `actual_function?` gave the right result, I was printing some values and I was expecting to see that test case pop out, but it never came, so I realized that `depth` was becoming negative.

I've added a failing test now. Let that be a valuable lesson of never fixing bugs without context.